app/configurator/components/filters.tsx

```diff
@@ -345,21 +348,43 @@ const MultiFilterContent = ({
     anchorEl?.focus();
   });
 
-  // Recomputes color palette making sure that used values
-  // are sorted first, so they have different colors
-  const handleRecomputeColorMapping = useEvent(
-    ({ random }: { random: boolean } = { random: false }) => {
-      const usedValues = new Set(values.map((v) => v.value));
-      dispatch({
-        type: "CHART_CONFIG_UPDATE_COLOR_MAPPING",
-        value: {
-          dimensionIri,
-          values: sortBy(colorComponent?.values, (d) =>
-            usedValues.has(d.value) ? 0 : 1
-          ),
-          random,
-        },
-      });
+  /** Handles updates to color mapping.
+   *
+   *  An update can be either a _reset_, to go back to original colors or a _shuffle_
+   * to create a new, randomized color mapping.
+   *
+   * @param type _reset_ or _shuffle_
+   */
+  const handleUpdateColorMapping = useEvent(
+    ({ type }: { type: "reset" | "shuffle" }) => {
+      const computedField = `${field}${
+        colorConfigPath !== undefined ? `.${colorConfigPath}` : ""
+      }`;
+      const values = colorComponent?.values || [];
+
+      switch (type) {
+        case "reset":
+          return dispatch({
+            type: "CHART_CONFIG_UPDATE_COLOR_MAPPING",
+            value: {
+              field: computedField,
+              dimensionIri,
+              values,
+              random: false,
+            },
+          });
+        case "shuffle":
+          const usedValues = new Set(values.map((v) => v.value));
+          return dispatch({
+            type: "CHART_CONFIG_UPDATE_COLOR_MAPPING",
+            value: {
+              field: computedField,
+              dimensionIri,
+              values: sortBy(values, (d) => (usedValues.has(d.value) ? 0 : 1)),
+              random: true,
+            },
+          });
```

Comment:
I wonder here why we do not have two events ?

Reply:
Two `useEvents` or two action types of reducer?

Reply:
two action types. This is a nit.